test/functional/rpc_blockchain.py

```
@@ -417,10 +417,58 @@
+        # Genesis block height estimate should return 0
+        hashes_per_second = self.nodes[0].getnetworkhashps(100, 0)
+        assert_equal(hashes_per_second, 0)
+
         # This should be 2 hashes every 2.6 minutes (156 seconds) or 1/78
+        hashes_per_second = self.nodes[0].getnetworkhashps()
         assert abs(hashes_per_second * 78 - 1) < 0.0001
 
+        # Test setting the first param of getnetworkhashps to -1 returns the average network
+        # hashes per second from the last difficulty change.
+        current_block_height = self.nodes[0].getmininginfo()['blocks']
+        blocks_since_last_diff_change = current_block_height % DIFFICULTY_ADJUSTMENT_INTERVAL + 1
```

Ruff (0.12.2)

Undefined name `DIFFICULTY_ADJUSTMENT_INTERVAL`

(F821)